src/jit/lsra.cpp

```diff
@@ -6373,6 +6387,24 @@ void LinearScan::unassignPhysReg(RegRecord* regRec, RefPosition* spillRefPositio
     {
         regRec->assignedInterval = regRec->previousInterval;
         regRec->previousInterval = nullptr;
+#ifdef _TARGET_ARM_
+        // Update overlapping floating point register for TYP_DOUBLE
+        if (regRec->assignedInterval->registerType == TYP_DOUBLE)
+        {
+            assert(isFloatRegType(regRec->registerType));
+
+            regNumber  nextRegNum = REG_NEXT(regRec->regNum);
+            RegRecord* nextRegRec = getRegisterRecord(nextRegNum);
+
+            nextRegRec->assignedInterval = nextRegRec->previousInterval;
+            nextRegRec->previousInterval = nullptr;
+
+            // TODO-ARM-Throughput: For ARM, we can reuse regRec->previousInterval,
+            // if following assertion is always true.
+            // Please take a look at tryAllocateFreeReg().
+            assert(nextRegRec->assignedInterval == regRec->assignedInterval);
```

Comment:
Thank you for feedback :)

The assumption in this PR is as follows.
For TYP_DOUBLE interval, we keep following mappings between a TYP_DOUBLE interval and two RegRecords which 
consist a double register.

For example, if we have assigned `d0` register (which is actually `f0` and `f1` in this LSRA ) to a interval `Interval_1` which is TYP_DOUBLE`, we keep following mappings.
```
When Interval_1->registerType == TYP_DOUBLE

// When a TYP_DOUBLE register is assigned to a TYP_DOUBLE interval
Interval_1->assginedReg = RegRecord_f0       // from interval to regrecord
RegRecord_f0->assignedInterval = Interval_1  // from regrecord to interval
RegRecord_f1->assignedInterval = Interval_1  // from regrecord to interval

// When double register is freed later
Interval_1->assignedReg = NULL
RegRecord_f0->previousInterval = Interval_1
RegRecord_f1->previousInterval = Interval_1
```

> I don't believe that this is a correct assert. One or both of the two registers may have been allocated to a TYP_FLOAT interval.

With above assumption, I think we can assume that two (single float) registers have been already assigned to a same TYP_DOUBLE interval, because  this assertion is inside `if (regRec->assignedInterval->registerType == TYP_DOUBLE)`.

And for the case where one or both register have been allocated to `TYP_FLOAT` interval after being freed for `TYP_DOUBLE` interval, both `regRec` and `nextRegRec` should have null `previousInterval` when the code reaches here. I think I've find and update the code where `previousInterval` is partially set. (Can you suggest where?)
 
What do you think of it?